numba/targets/rangeobj.py

```diff
@@ -230,5 +232,48 @@ def get(rnge):
         return get
 
 
+@register_jitable
+def impl_contains_helper(robj, val):
+    if robj.step > 0 and (val < robj.start or val >= robj.stop):
+        return False
+    elif robj.step < 0 and (val <= robj.stop or val > robj.start):
+        return False
+
+    return ((val - robj.start) % robj.step) == 0
+
+
+@overload(operator.contains)
+def impl_contains(robj, val):
+    def impl_false(robj, val):
+        return False
+
+    if not isinstance(robj, types.RangeType):
+        return
+
+    elif isinstance(val, (types.Integer, types.Boolean)):
+        return impl_contains_helper
+
+    elif isinstance(val, types.Float):
+        def impl(robj, val):
+            if val % 1 != 0:
+                return False
+            else:
+                return impl_contains_helper(robj, int(val))
+        return impl
+
+    elif isinstance(val, types.Complex):
```

Comment:
You might be able to get away with merging the `Float`  and `Complex` cases, but it might have a performance impact. Like this it contains some duplicated code, on the other hand merging the two might  make it slower for `Floats` as there is an additional check involved. Only `@timeit` will tell.

Reply:
I'm just going to leave them separate for now unless there is a strong preference for combining them.

Reply:
o.k.